scripts/deploy.ts

```diff
@@ -0,0 +1,284 @@
+import dotenv from 'dotenv'
+import fs from 'fs'
+import path from 'path'
+import readline from 'readline'
+import yargs from 'yargs'
+
+import { providers, utils, Wallet } from 'ethers'
+import {
+    AutomateSDK,
+    Secrets,
+    TaskTransaction,
+    TriggerConfig,
+    TriggerType,
+    Web3Function,
+    Web3FunctionUserArgs,
+} from '@gelatonetwork/automate-sdk'
+import { hideBin } from 'yargs/helpers'
+
+import { listDirectories, listJsonFiles } from './utils'
+
+dotenv.config()
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//// CONSTANTS & TYPES /////////////////////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const w3fImplementationsPath = './web3-functions'
+const w3fConfigsPath = './scripts/configs'
+const w3fIpfsDeploymentsPath = './scripts/pre-deployments.json'
+const abisPath = './abis'
+
+const domains = ['mainnet', 'gnosis']
+type Domain = (typeof domains)[number]
+
+const triggerTypeNames = ['block', 'cron', 'event', 'time']
+
+type BlockTrigger = {
+    type: 'block'
+}
+
+type CronTrigger = {
+    type: 'cron'
+    cron: string
+}
+
+type EventTrigger = {
+    type: 'event'
+    filter: {
+        address: string
+        topics: Array<{
```

Comment:
rename to something like eventInfo? Cause its used to parse info into `topics`

Reply:
done